WalletWasabi.Backend/Controllers/BatchController.cs

```diff
@@ -21,12 +21,13 @@ namespace WalletWasabi.Backend.Controllers;
 [Route("api/v" + Constants.BackendMajorVersion + "/btc/[controller]")]
 public class BatchController : ControllerBase
 {
-	public BatchController(BlockchainController blockchainController, ChaumianCoinJoinController chaumianCoinJoinController, HomeController homeController, OffchainController offchainController, Global global)
+	public BatchController(BlockchainController blockchainController, ChaumianCoinJoinController chaumianCoinJoinController, HomeController homeController, OffchainController offchainController, WabiSabiController wabiSabiController, Global global)
 	{
 		BlockchainController = blockchainController;
 		ChaumianCoinJoinController = chaumianCoinJoinController;
 		HomeController = homeController;
 		OffchainController = offchainController;
+		WabiSabiController = wabiSabiController;
```

Comment:
I would prefer to inject the `CoinJoinMempoolManager` directly here instead of depending on the controller.

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/10956#discussion_r1247550914

Reply:
Ok. This is not good then. `WabiSabiController` is for WabiSabi coordination and then it doesn't require a `CoinJoinMempoolManager`. Adding this an extra dependency makes the creation of instances more expensive and that's why you were forced to modify many UTs to add a dependency that they don't need/use. If you remember how painful writing the UTs for turbosync was, you should understand this is bad.

Another way to smell these things is by using this heuristic: if your dependency is used only in one method and only in one line of that method then it is wrong.

Anyway, feel free to ignore.

Reply:
At this point, I have to choose which direction to follow.

- nopara73: use the schema https://github.com/zkSNACKs/WalletWasabi/pull/10956#discussion_r1247550914
- lucas: only inject what requires.

I will keep it this way. It is the same design as we had in WW1. I believe having an `GetUnconfirmedCoinjoins` endpoint in `WabiSabiController` makes sense. Good for testing purposes in case of a mempool spike.